internal/zinc-core/src/main/java/xsbti/compile/TastyFiles.java

```diff
@@ -0,0 +1,23 @@
+package xsbti.compile;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Can be added to `IncOptions.auxiliaryClassFiles` so that the TASTy files are
+ * managed by the ClassFileManager
+ */
+public class TastyFiles implements AuxiliaryClassFiles {
+    @Override
+    public Path[] associatedFiles(Path classFile) {
+        if (classFile.endsWith(".class")) {
+            String absolutePath = classFile.toAbsolutePath().toString();
+            String prefix = absolutePath.substring(0, absolutePath.length() - 6);
+            Path tastyFile = Paths.get(URI.create(prefix + ".tasty"));
+            return new Path[] { tastyFile };
```

Comment:
This implementation indiscriminately returns a `Path` for a `.tasty` file even if it doesn't exist. Not all `.class` files have a corresponding `.tasty` file (only top-level classes have one). Is this expected, i.e., is the client supposed to ignored returned `Path`s that do not exist?

Reply:
Yes I think it is easier if the `ClassFileManager` is forgiving on auxiliary files that don't exist.
Also, as far as I can see, the `ClassFileManager` implementations are already forgiving on class files that don't exist.

I add a comment and a test case.